### ISSUE TITLE
lint: add long chain expression checker

### ISSUE THREAD

**Issue by Quasilyte** [PieselBois/kfulint#11]
Find long expression chains that are repeated more than N times (some kind of threshold) and consist of multiple selector/index expressions.

So, this code will trigger a warning:
```go
	type something struct {
		with struct {
			very struct {
				big struct {
					nesting struct {
						value int
					}
				}
			}
		}
	}
	var x something

	switch {
	case x.with.very.big.nesting.value > 0:
		fmt.Printf("positive value: %d\n", x.with.very.big.nesting.value)
	case x.with.very.big.nesting.value < 0:
		fmt.Printf("negative value: %d\n", x.with.very.big.nesting.value)
	default:
		fmt.Println("zero")
	}
```
And the alternative we are proposing:
```go
	value := x.with.very.big.nesting
	switch {
	case value > 0:
		fmt.Printf("positive value: %d\n", value)
	case value < 0:
		fmt.Printf("negative value: %d\n", value)
	default:
		fmt.Println("zero")
	}
```
Or, even better:
```go
	switch value := x.with.very.big.nesting; {
	case value > 0:
		fmt.Printf("positive value: %d\n", value)
	case value < 0:
		fmt.Printf("negative value: %d\n", value)
	default:
		fmt.Println("zero")
	}
```

We don't have to provide "fixed" code snippet.
The minimal goal is to alert the possibility for code improvement.
It means that in the simplest case it's OK to just report the problem, like:
> Expression chain `x.with.very.big.nesting.value` repeated 4 times, consider assigning it to local variable
